src/app-layer-dns-common.c

```diff
@@ -253,11 +253,11 @@ int DNSGetTxLogged(void *alstate, void *tx, uint32_t logger)
 
 /** \brief get value for 'complete' status in DNS
  *
- *  For DNS we use a simple bool. 1 means done.
+ *  For DNS 1 means request done, 2 means reply done (or lost).
  */
 int DNSGetAlstateProgressCompletionStatus(uint8_t direction)
 {
-    return 1;
+    return 2;
```

Comment:
On 11-07-16 20:36, Jason Ish wrote:
> In src/app-layer-dns-common.c
> <https://github.com/inliniac/suricata/pull/2172#discussion_r70312347>:
> 
>>   */
>>  int DNSGetAlstateProgressCompletionStatus(uint8_t direction)
>>  {
>> -    return 1;
>> +    return 2;
> 
> As all registration goes through functions, I think it would be easy to
> make -1 be the value that means assign it in OutputRegisterTxLogger,
> thus making 0 a valid value when registered WIthProgress().

Simple enough, lets try that.